synthetics-stats/src/entities/fees.ts

```diff
@@ -358,3 +360,178 @@ function getOrCreatePositionFeesInfoWithPeriod(
+    totalFees,
+    "1d",
+    transaction
+  );
+}
+
+export function handleMarketPoolValueUpdated(
+  eventData: EventData,
+  transaction: Transaction
+): void {
+  let poolValue = eventData.getIntItem("poolValue")!;
+  let marketAddress = eventData.getAddressItemString("market")!;
+
+  if (poolValue.equals(ZERO)) {
+    log.warning("Pool value is zero: {}", [marketAddress]);
+    return;
+  }
+
+  let poolValueRef = getPoolValue(marketAddress);
+
+  poolValueRef.poolValue = poolValue;
+
+  let pendingIds = poolValueRef.pendingCollectedMarketFeesInfoIds;
+  let fees = poolValueRef.pendingFeeUsds;
+
+  for (let i = 0; i < pendingIds.length; i++) {
+    let id = pendingIds[i];
+    let feeInfo = CollectedMarketFeesInfo.load(id);
+    let feeUsd = fees[i];
+
+    if (feeInfo) {
+      if (feeUsd) {
+        feeInfo.feeUsdPerPoolValue = getUpdatedFeeUsdPerPoolValue(
+          feeInfo!,
+          feeUsd,
+          poolValue
+        );
+
+        feeInfo.poolValue = poolValue;
+        feeInfo.save();
+      }
+    }
+  }
+
+  poolValueRef.pendingCollectedMarketFeesInfoIds = new Array<string>(0);
+  poolValueRef.pendingFeeUsds = new Array<BigInt>(0);
+
+  poolValueRef.save();
+}
+
+export function handlePositionImpactPoolDistributed(
+  eventData: EventData,
+  transaction: Transaction
+): void {
+  let event = new PositionImpactPoolDistributedEventData(eventData);
+  let market = MarketInfo.load(event.market);
+
+  if (!market) {
+    log.warning("Market not found: {}", [event.market]);
+    throw new Error("Market not found");
+  }
+
+  let indexToken = market.indexToken;
+  let tokenPrice = getTokenPrice(indexToken);
+  let amountUsd = event.distributionAmount.times(tokenPrice);
+  let poolValueRef = getPoolValue(event.market);
+
+  // 1h
+  let feesFor1h = getOrCreateCollectedMarketFees(
+    event.market,
+    transaction.timestamp,
+    "1h"
+  );
+
+  feesFor1h.feeUsdPerPoolValue = getUpdatedFeeUsdPerPoolValue(
+    feesFor1h,
+    amountUsd,
+    poolValueRef.poolValue
+  );
+
+  feesFor1h.feeUsdForPool = feesFor1h.feeUsdForPool.plus(amountUsd);
+  feesFor1h.poolValue = poolValueRef.poolValue;
+
+  feesFor1h.save();
+
+  // 1d
+  let feesFor1d = getOrCreateCollectedMarketFees(
+    event.market,
+    transaction.timestamp,
+    "1d"
+  );
+
+  feesFor1d.feeUsdPerPoolValue = getUpdatedFeeUsdPerPoolValue(
+    feesFor1d,
+    amountUsd,
+    poolValueRef.poolValue
+  );
+
+  feesFor1d.feeUsdForPool = feesFor1d.feeUsdForPool.plus(amountUsd);
+  feesFor1d.poolValue = poolValueRef.poolValue;
+
+  feesFor1d.save();
+
+  // total
+  let feesForTotal = getOrCreateCollectedMarketFees(
+    event.market,
+    transaction.timestamp,
+    "total"
+  );
+
+  feesForTotal.feeUsdPerPoolValue = getUpdatedFeeUsdPerPoolValue(
+    feesForTotal,
+    amountUsd,
+    poolValueRef.poolValue
+  );
+  feesForTotal.feeUsdForPool = feesForTotal.feeUsdForPool.plus(amountUsd);
+  feesForTotal.poolValue = poolValueRef.poolValue;
+
+  feesForTotal.save();
```

Comment:
at first glance it's repeated for 3 entities, does it make sense to refactor and create separate function?

Reply:
now only 2 entities, I think it's fine